### PR TITLE
Simplify code for Recordops.cs_pattern_of_constr

### PR DESCRIPTION
**Pull request by SkySkimmer**
This has different behaviour if called on an applied Rel, not sure if
that ever happens.

### CODE REVIEW

File: pretyping/recordops.ml [hunk 1]
```diff
@@ -197,32 +197,19 @@ let keep_true_projections projs kinds =
...
+    let patt, n, args = cs_pattern_of_constr env f in
+    patt, n, args @ Array.to_list vargs
+  | Rel n -> Default_cs, Some n, []
+  | Prod (_,a,b) when Vars.noccurn 1 b -> Prod_cs, None, [a; Vars.lift (-1) b]
+  | Proj (p, c) ->
+    let { Environ.uj_type = ty } = Typeops.infer env c in
+    let _, params = Inductive.find_rectype env ty in
+    Const_cs (ConstRef (Projection.constant p)), None, params @ [c]
+  | Sort s -> Sort_cs (Sorts.family s), None, []
+  | _ -> Const_cs (global_of_constr t) , None, []
```
Comment (mattam82) [inline pretyping/recordops.ml]: global_of_constr raises Not_found I suppose?

Comment (SkySkimmer) [reply pretyping/recordops.ml]: Yes

**Comment by mattam82** [comment]
This looks good to me, what do you think @gares ?

**Review by gares — APPROVED** [review]
thanks!

**Comment by SkySkimmer** [comment]
@mattam82 merge?

**Pull request merged**